### PR TITLE
Add NetworkPolicy for coredns

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/20946

I consider this a temporary hack before we know how we want to deal `coredns` https://github.com/giantswarm/coredns-app/pull/143 being a potential solution.

### CODE REVIEW

File: helm/cluster-resources/templates/coredns-np.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - ports:
+    - port: 1053
+      protocol: UDP
+    - port: 1053
+      protocol: TCP
+    - port: 53
+      protocol: UDP
+    - port: 53
+      protocol: TCP
+  - ports:
```
Comment (tfussell) [inline helm/cluster-resources/templates/coredns-np.yaml]: Why two separate blocks for these ports?

File: helm/cluster-resources/templates/coredns-np.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - port: 9153
+      protocol: TCP
+  podSelector:
+    matchLabels:
+      k8s-app: kube-dns
```
Comment (tfussell) [inline helm/cluster-resources/templates/coredns-np.yaml:25]: We might change this in the future. Could be nice to make it configurable.

File: helm/cluster-resources/templates/coredns-np.yaml (hunk 1 above)
Comment (kopiczko) [reply helm/cluster-resources/templates/coredns-np.yaml]: I copied the rendered manifest https://github.com/giantswarm/coredns-app/blob/master/helm/coredns-app/templates/np.yaml

File: helm/cluster-resources/templates/coredns-np.yaml (hunk 2 above)
Comment (kopiczko) [reply helm/cluster-resources/templates/coredns-np.yaml:25]: I don't consider this solution long-term so I don't really want to add all the bells and whistles.

**Review by tfussell — APPROVED** [review]

**Pull request merged**